plugins/snowflake/dbt/include/snowflake/macros/materializations/table.sql

```diff
@@ -49,51 +53,30 @@
   -- `BEGIN` happens here:
   {{ run_hooks(pre_hooks, inside_transaction=True) }}
 
-  -- build model
+  --build model
   {% call statement('main') -%}
-    {%- if non_destructive_mode -%}
-      {%- if old_relation is not none -%}
-        {{ create_table_as(create_as_temporary, intermediate_relation, sql) }}
-
-        {% set dest_columns = adapter.get_columns_in_relation(old_relation) %}
-        {% set dest_cols_csv = dest_columns | map(attribute='quoted') | join(', ') %}
-
-        insert into {{ target_relation }} ({{ dest_cols_csv }}) (
-          select {{ dest_cols_csv }}
-          from {{ intermediate_relation.include(database=(not create_as_temporary), schema=(not create_as_temporary)) }}
-        );
-      {%- else -%}
-        {{ create_table_as(create_as_temporary, target_relation, sql) }}
-      {%- endif -%}
-    {%- else -%}
-      {{ create_table_as(create_as_temporary, intermediate_relation, sql) }}
-    {%- endif -%}
-  {%- endcall %}
-
-  -- cleanup
-  {% if non_destructive_mode -%}
-    -- noop
-  {%- else -%}
+  -- we can leverage Snowflake create or replace table here to achieve an atomic replace.
     {% if old_relation is not none %}
+      {# -- I'm preserving one of the old checks here for a view, and to make sure Snowflake doesn't
```

Comment:
Can this logic be rewritten as 
```
{% if old_relation is not none and old_relation.type == 'view' %}
  {{ log("Dropping relation " ~ old_relation ~ " because it is a view and this model is a table.") }}
  {{ drop_relation_if_exists(old_relation) }}
{% endif %}
```

I think we don't need the nested if-statements anymore

Reply:
absolutely true. changed